libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp

```diff
@@ -235,6 +235,24 @@ int WiFiClientSecure::connect(const String& host, uint16_t port) {
   return connect(host.c_str(), port);
 }
 
+int WiFiClientSecure::connectAsync(const char* name, uint16_t port, uint8_t* waiting) {
+  if (waiting != NULL) *waiting = 0;
+  if (connected()) return 1;
+
+  IPAddress remote_addr;
+
+  if (!WiFi.hostByNameAsync(name, remote_addr, waiting)) {
```

Comment:
I don't think this fcn can work.  If `hostByNameAsync` is IN_PROGRESS, then it returns 0.  And so this routine will return failure if the async part actually takes effect.

Reply:
Done in 16d52717a6603ee422cb61c58741f518de233d19. Should work